### PR TITLE
Support SAML 2.0 as an auth backend

### PR DESCRIPTION
**Pull request by adnankobir**
## What does this PR accomplish?

Feature

addresses #1036

## Changes proposed by this PR:

[concourse/dex](https://github.com/concourse/dex) already includes the saml connector, this PR simply adds the skycmd flags required to enable saml authentication.

## Contributor Checklist

- [x] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [x] [Signed] all commits
- [ ] Added tests (Unit and/or Integration)
- [x] Updated [Documentation]
- [ ] Added release note (Optional)

## Reviewer Checklist

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

**Comment by taylorsilva** [comment]
Hey @adnankobir sorry about letting this PR fall through the cracks. We had quite a few people on the team go on vacation and we forgot to re-assign PR's for them. I'll un-assign the current engineer from this PR that way on Monday it'll be re-assigned to someone that's actually in the for the week.

File: skymarshal/skycmd/saml_flags.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	RegisterConnector(&Connector{
+		id:         "saml",
+		config:     &SAMLFlags{},
+		teamConfig: &SAMLTeamFlags{},
+	})
+}
+
+type SAMLFlags struct {
+	DisplayName        string      `long:"display-name" description:"The auth provider name displayed to users on the login page"`
+	SsoURL             string      `long:"sso-url" description:"(Required) SSO URL used for POST value"`
```
Comment (aoldershaw) [inline skymarshal/skycmd/saml_flags.go:22]: I notice that dex also supports `EntityIssuer` and `SSOIssuer` fields - not sure what they are for. Is there a reason you didn't include them?

**Comment by aoldershaw** [comment]
Update - I was able to get Concourse configured as a SAML app in Okta using the following configuration:

Screen Shot 2020-08-31 at 12 06 42 PM

...though admittedly I was just following the error messages/looking at Dex's implementation, and am not sure if everything is configured 100% correctly.

And then used the following docker-compose override:

```yaml
version: '3'

services:
  web:
    environment:
      CONCOURSE_SAML_DISPLAY_NAME: Test SAML
      CONCOURSE_SAML_SSO_URL: https://dev-X.okta.com/app/Y/Z/sso/saml
      CONCOURSE_SAML_CA_CERT: /src/okta.cert
```

The SSO URL and CA cert came from Okta under "View Setup Instructions". The setup instructions also include an `Identity Provider Issuer` field that looks like: `http://www.okta.com/...`, but I couldn't figure out what to do with it. Seemed to work without it, though.

---

@adnankobir would you be willing to document this feature in the docs (https://github.com/concourse/docs/tree/master/lit/docs/auth/configuring)?

File: skymarshal/skycmd/saml_flags.go (hunk 1 above)
Comment (adnankobir) [reply skymarshal/skycmd/saml_flags.go:22]: Thanks for pointing this out, I didn't include them as they were not required when I tested the okta integration, but for generic SAML use-cases someone might require these flags, so I suppose we can pass them through to dex

**Comment by adnankobir** [comment]
> Update - I was able to get Concourse configured as a SAML app in Okta using the following configuration:
> 
> Screen Shot 2020-08-31 at 12 06 42 PM
> 
> ...though admittedly I was just following the error messages/looking at Dex's implementation, and am not sure if everything is configured 100% correctly.
> 
> And then used the following docker-compose override:
> 
> ```yaml
> version: '3'
> 
> services:
>   web:
>     environment:
>       CONCOURSE_SAML_DISPLAY_NAME: Test SAML
>       CONCOURSE_SAML_SSO_URL: https://dev-X.okta.com/app/Y/Z/sso/saml
>       CONCOURSE_SAML_CA_CERT: /src/okta.cert
> ```
> 
> The SSO URL and CA cert came from Okta under "View Setup Instructions". The setup instructions also include an `Identity Provider Issuer` field that looks like: `http://www.okta.com/...`, but I couldn't figure out what to do with it. Seemed to work without it, though.
> 
> @adnankobir would you be willing to document this feature in the docs (https://github.com/concourse/docs/tree/master/lit/docs/auth/configuring)?

This is exactly how I have it setup as well, also didn't need the `Identity Provider Issuer` field. I'll update the docs and add the two optional flags. Thanks!

**Comment by adnankobir** [comment]
@aoldershaw updated docs and added the two optional flags

**Pull request merged**